src/errorhandling/ErrorHandler.ts

```diff
@@ -17,90 +17,51 @@ export enum SystemEvent {
   ExtrinsicFailed = '0x0001',
 }
 
-export class ErrorHandler {
-  private static readonly ERROR_MODULE_NAME = 'Error'
+const ERROR_MODULE_NAME = 'error'
 
-  /**
-   * [STATIC] Checks if there is `SystemEvent.ExtrinsicFailed` in the list of
-   * transaction events within the given `extrinsicResult`.
-   *
-   * @param extrinsicResult The result of a submission.
-   * @returns Whether the extrinsic submission failed.
-   */
-  public static extrinsicFailed(extrinsicResult: SubmittableResult): boolean {
-    const events: EventRecord[] = extrinsicResult.events || []
+/**
+ * Checks if there is `SystemEvent.ExtrinsicFailed` in the list of
+ * transaction events within the given `extrinsicResult`.
+ *
+ * @param extrinsicResult The result of a submission.
+ * @returns Whether the extrinsic submission failed.
+ */
+export function extrinsicFailed(extrinsicResult: SubmittableResult): boolean {
```

Comment:
I think you will have to explain to me, why the errorhandler shrinked

Reply:
> My only concern would be the inability to wait for both `IN_BLOCK` and `FINALIZED`. 

Doesn't `FINALIZED` already imply `IN_BLOCK`?

Reply:
I think we are talking about individual promises that resolve at different stages. Should be possible, I'm working on something right now